add-chain/go.mod

```diff
@@ -0,0 +1,33 @@
+module github.com/ethereum-optimism/superchain-registry/add-chain
+
+go 1.21.9
```

Comment:
Does each component have it's own go.mod file? It seems a bit easier to only have one.

Reply:
One advantage of having separate modules is that it can avoid dependency cycles. The `add-chain` module is free to depend on `go-ethereum`, whereas doing that inside `superchain` is probably a problem since `op-geth` depends on `superchain`. https://www.notion.so/oplabs/Superchain-registry-module-structure-4b81c28b6a1e4e169056378eacdd9f14